File: proyectos_c#/juegos/JuegoSnake/JuegoSnake/Comida.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace JuegoSnake
{
    public class Comida: Objeto
    {
        public Comida(int tamano): base(tamano){
            this.colocar();
        }
        public void colocar(){
            this.x = this.generarx();
            this.y = this.generary();
        }

        public void dibujar(Graphics g)
        {
            Pen bluePen = new Pen(Color.Blue, 3);
            g.DrawRectangle(bluePen, this.x, this.y, this.tamano, this.tamano);
        }
        public int generarx() {
            //entre 0 y 500
            //Size(369, 262)
            int tam = 369 / 10;
            Random random = new Random();
            int num = random.Next(0, tam) * 10;// Math.floor(Math.random() * tamx) * 10;
            return num;
        }
        public int generary(){
            //entre 0 y 500
            int tam = 262 / 10;
            Random random = new Random();
            int num = random.Next(0, tam) * 10;// Math.floor(Math.random() * tamx) * 10;
            return num;
        }
    }
}

File: proyectos_basic/juegos/SnakeBasic/SnakeBasic/Comida.vb
Public Class Comida
    Inherits Objeto
    Public Sub New(tamano As Integer)
        MyBase.New(tamano)
        Me.colocar()
    End Sub
    Public Sub colocar()
        Me.x = Me.generarx()
        Me.y = Me.generary()
    End Sub
    Public Function generarx()
        'entre 0 y 500
        'Size(567, 285)
        Dim tam As Integer = 467 / 10
        Dim random As Random = New Random()
        Dim num As Integer = random.Next(0, tam) * 10
        Return num
    End Function
    Public Function generary()
        'entre 0 y 500
        'Size(567, 285)
        Dim tam As Integer = 285 / 10
        Dim random As Random = New Random()
        Dim num As Integer = random.Next(0, tam) * 10
        Return num
    End Function
    Public Sub dibujar(g As Graphics)
        Dim bluePen As Pen = New Pen(Color.Blue, 3)
        g.DrawRectangle(bluePen, Me.x, Me.y, Me.tamano, Me.tamano)
    End Sub
End Class
